pallets/thea/src/lib.rs

```diff
@@ -229,84 +297,41 @@ pub mod pallet {
-			bls_signature: [u8; 96],
-			payload: Payload<T::AccountId>,
+			network_id: u8,
+			bls_key: [u8; 192],
+			who: T::AccountId,
 		) -> DispatchResult {
-			ensure_signed(origin)?;
-			ensure!(payload.amount > 0, Error::<T>::AmountCannotBeZero);
-			// Fetch Deposit Nonce
-			let nonce = <DepositNonce<T>>::get(payload.network_id.saturated_into::<Network>());
-			ensure!(payload.deposit_nonce == nonce + 1, Error::<T>::DepositNonceError);
-			// Ensure assets are registered
-			ensure!(
-				asset_handler::pallet::TheaAssets::<T>::contains_key(payload.asset_id),
-				Error::<T>::AssetNotRegistered
-			);
+			ensure_root(origin)?;
 
-			// Fetch current active relayer set BLS Keys
-			let current_active_relayer_set = Self::get_relayers_key_vector(payload.network_id);
-
-			// Call host function with current_active_relayer_set, signature, bit_map, verify nonce
-			ensure!(
-				thea_primitives::thea_ext::bls_verify(
-					&bls_signature,
-					bit_map,
-					&payload.encode(),
-					&current_active_relayer_set
-				),
-				Error::<T>::BLSSignatureVerificationFailed
-			);
+			// Fetch Storage
+			let mut current_bls = Self::get_relayers_key_vector(network_id);
+			let mut current_ecdsa = Self::get_auth_list(network_id);
+			let key = BLSPublicKey(bls_key);
 
-			// Update deposit Nonce
-			<DepositNonce<T>>::insert(payload.network_id.saturated_into::<Network>(), nonce + 1);
+			// Update Storage
+			current_bls.try_push(key).unwrap();
+			<RelayersBLSKeyVector<T>>::insert(network_id, current_bls);
+			current_ecdsa.try_push(who.clone()).unwrap();
+			<AuthorityListEcdsa<T>>::insert(network_id, current_ecdsa);
 
-			// Update Storage item
-			let approved_deposit = ApprovedDeposit {
-				asset_id: payload.asset_id,
-				amount: payload.amount,
-				tx_hash: payload.tx_hash,
-			};
-			// TODO[#610]: Will be refactored in integration with Staking
-			if <ApprovedDeposits<T>>::contains_key(&payload.who) {
-				<ApprovedDeposits<T>>::mutate(payload.who.clone(), |bounded_vec| {
-					if let Some(inner_bounded_vec) = bounded_vec {
-						inner_bounded_vec.try_push(approved_deposit).unwrap();
-					}
-				});
-			} else {
-				let mut my_vec: BoundedVec<ApprovedDeposit, ConstU32<100>> = Default::default();
-				if let Ok(()) = my_vec.try_push(approved_deposit) {
-					<ApprovedDeposits<T>>::insert::<
-						T::AccountId,
-						BoundedVec<ApprovedDeposit, ConstU32<100>>,
-					>(payload.who.clone(), my_vec);
-					<AccountWithPendingDeposits<T>>::mutate(|accounts| {
-						accounts.insert(payload.who.clone())
-					});
-				}
-			}
+			Ok(())
+		}
 
-			// Emit event
-			Self::deposit_event(Event::<T>::DepositApproved(
-				payload.network_id,
-				payload.who,
-				payload.asset_id,
-				payload.amount,
-				payload.tx_hash,
-			));
+		///Approve Deposit
+		#[pallet::weight(1000)]
+		pub fn approve_deposit(
+			origin: OriginFor<T>,
+			bit_map: u128,
+			bls_signature: [u8; 96],
+			token_type: TokenType,
+			payload: Vec<u8>,
+		) -> DispatchResult {
```

Comment:
Please move back parameters definition within the dockblock which removed in this PR for this method.